packages/orchestrator-ui-components/src/components/WFOPageTemplate/WFOSidebar/WFOSidebar.tsx

```diff
@@ -17,11 +17,17 @@ import { WFOStartCreateWorkflowButtonComboBox } from './WFOStartCreateWorkflowBu
 
 export const WFOSidebar: FC = () => {
     const router = useRouter();
+    const [isSideNavOpenOnMobile, setisSideNavOpenOnMobile] = useState(false);
+
+    const toggleOpenOnMobile = () => {
+        setisSideNavOpenOnMobile(!isSideNavOpenOnMobile);
```

Comment:
Toggling like this can have unexpected results since the operation to set isSideNavOpenOnMobile might not be finished yet - for example when doing a few in a row.  When setting a value you get the current state so this is my suggestion:

```suggestion
        setisSideNavOpenOnMobile((openState) => !openState);
```

Reply:
We should actually apply this way in our whole project indeed. Also, officially the app is not supported on mobile. Do we really want to add extra logic specific for this?

Reply:
No callback? (if you are thinking about better ways; that should be the way forward, I suppose)

Reply:
What do you mean by no callback?

Reply:
E.g.:

```
    const toggleMobile = useCallback(() => {
        setIsSideNavOpenOnMobile((openState) => !openState);
    }, [setIsSideNavOpenOnMobile])
```
And

```
        <EuiSideNav
            mobileTitle="Main menu"
            toggleOpenOnMobile={toggleMobile}
            isOpenOnMobile={isSideNavOpenOnMobile}
            item=[]
            />          
```

Reply:
My comment was about perusing the possibility of using a function inside of the setState function that gets the current state passed into it instead of using the state value directly to avoid weird errors that result from the order in which React does state updates which are not always sequential. Using or not using arrow function is a valid topic, in this case it doesn't really matter though. The point of doing a state update is to trigger a rerender.

Reply:
Yes; I agreee -> 2 different situations.

removed the callback construction: as it was just to illustrate my point of view.

1. We have a lot of these arrow functions inside JSX props in our code base
2. We have a lot of function calls who rely on sequential state handling

I left a couple of comments open; in the hope to come up with some rules; it feels like a waste of time that we have to re-discuss these things every PR.